prov/rxd/src/rxd_ep.c

```diff
@@ -421,11 +421,6 @@ ssize_t rxd_ep_post_data_pkts(struct rxd_ep *ep, struct rxd_x_entry *tx_entry)
 		if (!pkt_entry)
 			return -FI_ENOMEM;
 
-		if (tx_entry->op == RXD_DATA_READ && !tx_entry->bytes_done) {
-			tx_entry->start_seq = ep->peers[tx_entry->peer].tx_seq_no;
-			ep->peers[tx_entry->peer].tx_seq_no = tx_entry->start_seq +
-							      tx_entry->num_segs;
-		}
```

Comment:
If the goal was to move this check outside of the while loop, why not just move it up inside this function?

Reply:
rxd_ep_post_data_pkts gets called in multiple locations but only one of those places deals with DATA_READ packets. We can move it within the same function - it would still work and help but it would also still keep that unnecessary check in the places that don't need them.